addons/assignGear/functions/fnc_assignGearMan.sqf

```diff
@@ -60,18 +59,22 @@ if (isText (_path >> "init")) then {
 };
 
 _unit setVariable [QGVAR(gearSetup), true, true];
-_unit setVariable ["F_Gear_Setup", true, true]; //TODO: legacy variable sync for radios, remove eventually
 
-#ifdef DEBUG_MODE_FULL
-    private _runTime = diag_tickTime - _startTime;
+END_COUNTER(assignGearMan);
 
-    if (isNil QGVAR(agmRunCount)) then {
-        GVAR(agmRunCount) = 0;
-        GVAR(agmRunTime) = 0;
-    };
 
-    INC(GVAR(agmRunCount));
-    GVAR(agmRunTime) = GVAR(agmRunTime) + _runTime;
-    diag_log format ["[POTATO-assignGear] - assignGearMan ran for class %1, with a run time of %2", _loadout, _runTime];
-    diag_log format ["[POTATO-assignGear] - assignGearMan ran %1 times, with an average run time of %2", GVAR(agmRunCount), GVAR(agmRunTime) / GVAR(agmRunCount)];
-#endif
+// Stupid hack to fix gear automaticly until we can sort out why setUnitLoadout ocasionly fails on JIPs
+[{
+    params ["_unit", "_loadoutArray"];
+    if (isNull _unit) exitWith {};
+    
+    private _arrayUniform = (_loadoutArray select 3) param [0, ""];
+    private _arrayVest = (_loadoutArray select 4) param [0, ""];
+    INFO_4("Checking loadout: Uniform [%1-%2] Vest [%3-%4]",_arrayUniform,uniform _unit,_arrayVest,vest _unit);
+
+    if ((_arrayUniform != (uniform _unit)) || {_arrayVest != (vest _unit)}) then {
+        ERROR_2("Mismatch [%1] [%2]",name _unit, typeOf _unit);
+        _unit setUnitLoadout _loadoutArray;
+        ["potato_adminMsg", [(format ["Auto-reset gear on %1", name _unit]), "Server"]] call CBA_fnc_globalEvent;
+    };
+}, [_unit, _loadoutArray], 15] call CBA_fnc_waitAndExecute
```

Comment:
waitAndExecute is sorted and only 1 will be checked each frame, so it's not too bad
But it also makes sorting more expensive... either way it shouldn't matter to much